### ISSUE TITLE
Eclipse crash

### ISSUE THREAD

**Issue by ascob** [i-rinat/freshplayerplugin#298]
When trying to use autocomplete in Eclipse (Ctrl+Space) it crashes.
Seems to be the same issue as #207 
```
NOT SANDBOXED
[fresh 15601] [NP] NP_Initialize aNPNFuncs=0x7fcd5c105168, aNPPFuncs=0x7fcd5c1050c8, browser API version = 19
[fresh 15711]    calling callback={.func=0x7fcd00acc020, .user_data=0x7fcdf9e43f78, .flags=0}, result=0, origin=ppb_message_loop_post_work
[fresh 15711]    returning from callback={.func=0x7fcd00acc020, .user_data=0x7fcdf9e43f78, .flags=0}, result=0, origin=ppb_message_loop_post_work
[fresh 15601] do_load_ppp_module, can't open /opt/google/chrome/PepperFlash/libpepflashplayer.so
[fresh 15601] do_load_ppp_module, can't open /opt/google/chrome-beta/PepperFlash/libpepflashplayer.so
[fresh 15601] do_load_ppp_module, can't open /opt/google/chrome-unstable/PepperFlash/libpepflashplayer.so
libva info: VA-API version 0.36.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_36
libva info: va_openDriver() returns 0
[fresh 15601] libva version 0.36
[fresh 15601] libva driver vendor: Intel i965 driver for Intel(R) Ironlake Mobile - 1.4.1.pre1 (1.4.1.pre1)
[fresh 15601] GLX version 1.4
[fresh 15601] screensavers found: fd.o-screensaver gnome-screensaver
[fresh 15601] found XRender
[fresh 15601]    calling callback={.func=0x7fcd00ac7740, .user_data=0x7fcdbc27b1a0, .flags=0}, result=0, origin=call_plugin_init_module
[fresh 15601]    returning from callback={.func=0x7fcd00ac7740, .user_data=0x7fcdbc27b1a0, .flags=0}, result=0, origin=call_plugin_init_module
[fresh 15711]    calling callback={.func=0x7fcd00ac77c0, .user_data=0x7fcdbc27b1a0, .flags=0}, result=0, origin=call_plugin_init_module_prepare_comt
[ERROR:flash/platform/pepper/common/pep_process.cpp(19)] CHECK(g_process_type == PROCESS_TYPE_UNKNOWN) failed
```

**Comment by i-rinat** [comment]
Managed to reproduce crash on initial workspace creation in Eclipse. However that could be a different issue, since I haven't seen `[ERROR:flash/platform/pepper/common/pep_process.cpp(19)] CHECK(g_process_type == PROCESS_TYPE_UNKNOWN) failed` line.

**Comment by i-rinat** [comment]
Well, I've pushed changes I was planned in 8d95efa983da3e1cb0576ca9c374bc9f5f280d9e. But now I realize that log you posted have only `NP_Initialize` call, without preceding `NP_GetValue` or `NP_GetPluginVersion`. So the patch probably won't have any effect.

I wonder what is `g_process_type` and why it's equal to `PROCESS_TYPE_UNKNOWN`. Unfortunately, source code is not available.

**Comment by i-rinat** [comment]
Found the reason for Eclipse crashing in my case. I've recently added signal handlers to automate backtrace collection. But they were added unconditionally, with overwriting existing handlers. And it also turned out, Java runtime routinely uses signals, with SIGSEGV amongst them, to speedup Null-pointer checks.

b384a09716b5fa7582133d196e1cf15669dd0c5c adds checks for existing handlers, and prevents installation of own handler if there is one already set.

**Comment by ascob** [comment]
Unfortunately I still get the same error.

**Comment by ivoshm** [comment]
@i-rinat: Good work, I had similar problem with Bonita (Eclipse based workflow editor) and with last patch disappeared.

**Comment by i-rinat** [comment]
@ascob,
Yeah, I know. What fixed was a different bug. It was introduced not so long ago (there is now a code which launches `gdb` against failed process to collect information about crash). So it could be that buggy code didn't reach you, especially if you are not using the most recent version of freshwrapper from the repository.

Bug with triggering assertion of "g_process_type == PROCESS_TYPE_UNKNOWN" is still there.

That string originates from `libpepflashplayer.so` itself. And since I don't have its source code, I can only guess what checks it performs, what do they mean, and why do they fail. I'm currently out of ideas.

**Comment by i-rinat** [comment]
Investigated PepperFlash code a bit. From what I've found, code that outputs message 
"CHECK(g_process_type == PROCESS_TYPE_UNKNOWN) failed" is called from either `PPP_InitializeModule` or from `PPP_InitializeBroker`. I don't use latter, so only first remains. Also there is a guard variable which prevents assertion if `PPP_InitializeModule` called first time.

I've tried to make two make consecutive calls of `PPP_InitializeModule` and got the same message. Aside artificial provoking of the bug, there should be no multiple calls `PPP_InitializeModule` since there is a guard variable in `NP_Initialize` which prevents calling it twice.

So I'm confused. It should not happen, but happens.
